pinotdb/db.py

```diff
@@ -161,12 +201,23 @@ def execute(self, operation, parameters=None):
         cursor = self.cursor()
         return cursor.execute(operation, parameters)
 
+    @check_closed
+    async def execute_async(self, operation, parameters=None):
```

Comment:
From the usage perspective, since we put all the methods in the same class, then do you think users can use the same connection for mixed sync and async execution?

If not, we should add the checks. E.g. when the user call execute() or close() for an async connection, the client should just error out.

Another thought here is to make a new Class `AyncConnection` extend `Connection`, then implement `execute_async` and override `execute` to throw an error.